Mathlib/Data/Set/Disjoint.lean

```diff
@@ -111,6 +107,20 @@ alias ⟨_, _root_.HasSubset.Subset.disjoint_compl_left⟩ := disjoint_compl_lef
 
 alias ⟨_, _root_.HasSubset.Subset.disjoint_compl_right⟩ := disjoint_compl_right_iff_subset
 
+theorem diff_union_diff_cancel_of_inter_subset_of_subset_union (hi : s ∩ u ⊆ t)
+    (hu : t ⊆ s ∪ u) : (s \ t) ∪ (t \ u) = s \ u := by
```

Comment:
unless there's another theorem you think should get this name, I would prefer:
```suggestion
theorem diff_union_diff_cancel (hi : s ∩ u ⊆ t) (hu : t ⊆ s ∪ u) :
    (s \ t) ∪ (t \ u) = s \ u := by
```

Reply:
Unfortunately this one is taken. Do you think a primed version is maybe better?

Reply:
I've primed it, and changed the proof so it can sit right next to the unprimed version.